### ISSUE TITLE
vector cuda returns float64

### ISSUE THREAD

**Issue by adler-j** [odlgroup/odl#108]
When running `vector` with `impl='cuda'` you get a float64 result, should be float32

```python
>>> vector([1.0, 2.0, 3.0], impl='cuda').dtype
dtype('float64')
```

Edit, specifically this test fails:

```sh
    def test_vector_cuda():
    
        # Rn
        inp = [1.0, 2.0, 3.0]
    
        x = vector(inp, impl='cuda')
        assert isinstance(x, odl.CudaRnVector)
>       assert x.dtype == np.dtype('float32')
E       assert dtype('float64') == dtype('float32')
```

**Comment by kohr-h** [comment]
This happens since `vector` fully relies on `numpy.array` to determine the "correct" data type, which happens to be `float64` for floating point data.

To get around this, we need to be more specific about the default data types for the spaces we implement. Not sure how do this in a smart way, however. It would be nice if one could tell NumPy which data type to use by default. I'll do a bit research on this. If it turns out to be infeasible, we probably need to do `arr.astype('float32')` in the CUDA case and live with the copy.

**Comment by adler-j** [comment]
Closed by 47a70da422f0b3d93a581532528c9957ca86a510
